modules/core/src/lib/deck-picker.js

```diff
@@ -26,31 +26,122 @@ import {getClosestObject, getUniqueObjects} from '../utils/query-object';
+    this.pickingEvent = null;
+    this.lastPickedInfo = {
+      // For callback tracking and autohighlight
+      index: -1,
+      layerId: null,
+      info: null
+    };
+  }
 
-  // Pick the closest object at the given (x,y) coordinate
-  pickObject(
-    gl,
-    {
+  // Returns a new picking info object by assuming the last picked object is still picked
+  getLastPickedObject({x, y, viewports}) {
+    const layers = this.layerManager.getLayers();
+    const lastPickedInfo = this.lastPickedInfo.info;
+    const lastPickedLayerId = lastPickedInfo && lastPickedInfo.layer && lastPickedInfo.layer.id;
+    const layer = lastPickedLayerId ? layers.find(l => l.id === lastPickedLayerId) : null;
+    const coordinate = viewports[0] && viewports[0].unproject([x, y]);
+
+    const info = {
+      x,
+      y,
+      coordinate,
+      // TODO remove the lngLat prop after compatibility check
+      lngLat: coordinate,
+      layer
+    };
+
+    if (layer) {
+      return Object.assign({}, lastPickedInfo, info);
+    }
+    return Object.assign(info, {color: null, object: null, index: -1});
+  }
+
+  // Pick the closest info at given coordinate
+  pickObject({
+    x,
+    y,
+    mode,
+    radius = 0,
+    layerIds,
+    viewports,
+    useDevicePixels,
+    depth = 1,
+    event = null
+  }) {
+    // Allow layers to access the event
+    this.pickingEvent = event;
+
+    const layers = this.layerManager.getLayers({layerIds});
+    const {layerFilter, activateViewport} = this.layerManager;
+
+    const result = this.pickClosetObject({
+      // User params
+      x,
+      y,
+      radius,
       layers,
+      mode,
+      layerFilter,
+      depth,
+      // Injected params
       viewports,
+      onViewportActive: activateViewport,
+      pickingFBO: this.layerManager.getPickingBuffer(),
+      lastPickedInfo: this.lastPickedInfo,
+      useDevicePixels
+    });
+
+    // Clear the current event
+    this.pickingEvent = null;
+    return result;
+  }
+
+  // Get all unique infos within a bounding box
+  pickObjects({x, y, width, height, layerIds, viewports, useDevicePixels}) {
+    const layers = this.layerManager.getLayers({layerIds});
+    const {layerFilter, activateViewport} = this.layerManager;
+
+    return this.pickVisibleObjects({
       x,
       y,
-      radius,
+      width,
+      height,
+      layers,
       layerFilter,
-      depth = 1,
-      mode,
-      onViewportActive,
-      pickingFBO,
-      lastPickedInfo,
+      mode: 'pickObjects',
+      viewports,
+      onViewportActive: activateViewport,
+      pickingFBO: this.layerManager.getPickingBuffer(),
       useDevicePixels
-    }
-  ) {
+    });
+  }
+
+  // Private
+  // Pick the closest object at the given (x,y) coordinate
+  pickClosetObject({
+    layers,
+    viewports,
+    x,
+    y,
+    radius,
+    layerFilter,
+    depth = 1,
+    mode,
+    onViewportActive,
+    pickingFBO,
+    lastPickedInfo,
+    useDevicePixels
```

Comment:
Same TODO item as the `DeckRenderer`: `useDevicePixels` needs to be handled in `setProps` instead of passing around like this.